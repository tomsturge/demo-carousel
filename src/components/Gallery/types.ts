export interface ImageObj {
  src: string;
  alt: string;
}
