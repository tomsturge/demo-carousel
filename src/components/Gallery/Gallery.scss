.carousel {
  height: 770px;
  width: 100%;
  position: relative;
  overflow: hidden;
  margin: 0 auto;
}

.carousel__track {
  margin: 0;
  padding: 0;
  list-style: none;
  width: 100%;
  display: flex;
  position: absolute;
  left: -100%;

  &.animate {
    transition: left .3s;
  }

  img {
    max-width: 1200px;
  }
}

.carousel__arrow {
  position: relative;
  z-index: 10;
}

.carousel__control {
  position: absolute;
  bottom: 1em;
  width: 100%;
  text-align: center;
}

.carousel__button {
  cursor: pointer;
}
