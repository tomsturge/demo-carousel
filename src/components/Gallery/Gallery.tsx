import React from 'react';
import { connect } from 'react-redux'

const Gallery: React.FunctionComponent = () => 
<>
  hi
</>;

export default connect()(Gallery);
