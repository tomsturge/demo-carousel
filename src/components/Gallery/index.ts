export { default as Gallery } from './';
