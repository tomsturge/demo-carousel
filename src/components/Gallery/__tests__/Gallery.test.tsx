import * as React from 'react';
import { render, cleanup } from '@testing-library/react';

import Gallery from '../Gallery';

afterEach(cleanup);

describe('Gallery', () => {
  it('should display the content', () => {
    const { asFragment } = render(<Gallery />);
    expect(asFragment()).toMatchSnapshot();
  });
});
