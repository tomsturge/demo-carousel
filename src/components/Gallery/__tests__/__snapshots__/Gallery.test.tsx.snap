// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should display the content 1`] = `
<DocumentFragment>
  <div
    class="carousel"
  >
    <div
      class="carousel__track animate"
      style="left: -100%;"
    >
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="10.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="01.jpg"
      />
      <img
        alt="Ground level shot of a misty forest looking up at the milky yellow sky"
        src="02.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="03.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="04.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="05.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="06.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="07.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="08.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="09.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="10.jpg"
      />
      <img
        alt="Macro of dandelion seeds covered in morning dew "
        src="01.jpg"
      />
    </div>
    <button
      class="carousel__arrow"
    >
      Prev
    </button>
    <button
      class="carousel__arrow"
    >
      Next
    </button>
    <div
      class="carousel__control"
    >
      <input
        checked=""
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
      <input
        class="carousel__button"
        name="control"
        type="radio"
      />
    </div>
  </div>
</DocumentFragment>
`;
